modules/scylladb/scylladb_test.go

```diff
@@ -0,0 +1,195 @@
+		session, err := cluster.CreateSession()
+		require.NoError(t, err)
+		defer session.Close()
+
+		var address string
+		err = session.Query("SELECT address FROM system.clients").Scan(&address)
+		require.NoError(t, err)
+	})
+}
+
+func TestScyllaWithConfig(t *testing.T) {
+	ctx := context.Background()
+
+	ctr, err := scylladb.Run(ctx,
+		"scylladb/scylla:6.2",
+		scylladb.WithConfig(bytes.NewReader(scyllaYaml)),
+		scylladb.WithShardAwareness(),
+	)
+	require.NoError(t, err)
+	testcontainers.CleanupContainer(t, ctr)
+
+	t.Run("test-without-shard-awareness", func(t *testing.T) {
+		host, err := ctr.ConnectionHost(ctx, 9042)
+		require.NoError(t, err)
+
+		cluster := gocql.NewCluster(host)
+		session, err := cluster.CreateSession()
+		require.NoError(t, err)
+		defer session.Close()
+
+		var clusterName string
+		err = session.Query("SELECT cluster_name FROM system.local").Scan(&clusterName)
+		require.NoError(t, err)
+
+		// the "Amazing ScyllaDB Test" cluster name is set in the scylla.yaml file
+		require.Equal(t, "Amazing ScyllaDB Test", clusterName)
+	})
+
+	t.Run("test-with-shard-awareness", func(t *testing.T) {
+		host, err := ctr.ConnectionHost(ctx, 19042)
+		require.NoError(t, err)
+
+		cluster := gocql.NewCluster(host)
+		session, err := cluster.CreateSession()
+		require.NoError(t, err)
+		defer session.Close()
+
+		var clusterName string
+		err = session.Query("SELECT cluster_name FROM system.local").Scan(&clusterName)
+		require.NoError(t, err)
+
+		// the "Amazing ScyllaDB Test" cluster name is set in the scylla.yaml file
+		require.Equal(t, "Amazing ScyllaDB Test", clusterName)
+	})
+}
+
+func TestScyllaAlternator(t *testing.T) {
+	ctx := context.Background()
+
+	alternatorPort := uint16(8000)
+
+	t.Run("test-with-alternator", func(t *testing.T) {
+		ctr, err := scylladb.Run(ctx,
+			"scylladb/scylla:6.2.2",
+			scylladb.WithAlternator(alternatorPort),
+		)
+		require.NoError(t, err)
+		testcontainers.CleanupContainer(t, ctr)
+
+		client, err := getDynamoAlternatorClient(t, ctr, alternatorPort)
+		require.NoError(t, err)
+		requireCreateTable(t, client)
+	})
+
+	t.Run("test-without-alternator", func(t *testing.T) {
+		ctr, err := scylladb.Run(ctx,
+			"scylladb/scylla:6.2",
+		)
+		require.NoError(t, err)
+		testcontainers.CleanupContainer(t, ctr)
+
+		_, err = getDynamoAlternatorClient(t, ctr, alternatorPort)
+		require.Error(t, err)
+	})
+}
+
+type scyllaAlternatorResolver struct {
+	HostPort string
+}
+
+func (r *scyllaAlternatorResolver) ResolveEndpoint(_ context.Context, _ dynamodb.EndpointParameters) (smithyendpoints.Endpoint, error) {
+	return smithyendpoints.Endpoint{
+		URI: url.URL{Host: r.HostPort, Scheme: "http"},
+	}, nil
+}
+
+func getDynamoAlternatorClient(t *testing.T, c *scylladb.Container, port uint16) (*dynamodb.Client, error) {
+	t.Helper()
+	var errs []error
+
+	hostPort, err := c.ConnectionHost(context.Background(), port)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("get connection string: %w", err))
```

Comment:
suggestion: describe the method
```suggestion
		errs = append(errs, fmt.Errorf("connection host: %w", err))
```